app/src/main/java/org/simple/clinic/newentry/PatientEntryEffectHandler.kt

```diff
@@ -105,7 +105,13 @@ class PatientEntryEffectHandler @AssistedInject constructor(
           .flatMapSingle { getPatientEntryAndFacility }
           .subscribeOn(scheduler)
           .map { (entry, facility) ->
-            entry.takeIf { it.address != null } ?: entry.withAddress(Address.withDistrictAndState(facility.district, facility.state))
+            if (entry.address != null && entry.address.doesNotHaveDistrictAndState) {
```

Comment:
This change is unrelated to this PR, please extract it to a different PR.

Reply:
I made this change part of this PR because when we prefilled the data then we added data to the `colonyOrVillage` field therefore making `entry.address != null` and the district and state fields to not being prefilled. Which kinda broke the previous flow.